File: src/models/Notification.ts
import type { NotificationType, Role } from "@/types/auth";

export type NotificationRow = {
  id: string;
  user_role: Role;
  user_id: string;
  title: string;
  message: string;
  type: NotificationType;
  link: string | null;
  is_read: boolean;
  created_at: string;
};

export type Notification = {
  id: string;
  userRole: Role;
  userId: string;
  title: string;
  message: string;
  type: NotificationType;
  link: string | null;
  isRead: boolean;
  createdAt: string;
};

File: rkispro-android/app/src/main/java/com/rkispro/marketplace/data/model/Notification.kt
package com.rkispro.marketplace.data.model

import com.google.gson.annotations.SerializedName

data class AppNotification(
    val id: String,
    @SerializedName("user_role") val userRole: String,
    @SerializedName("user_id") val userId: String,
    val title: String,
    val message: String,
    val type: String = "info",
    val link: String? = null,
    @SerializedName("is_read") val isRead: Boolean = false,
    @SerializedName("created_at") val createdAt: String? = null
)
